kernel/src/engine/arrow_conversion.rs

```diff
@@ -19,19 +19,61 @@ pub(crate) const MAP_ROOT_DEFAULT: &str = "key_value";
 pub(crate) const MAP_KEY_DEFAULT: &str = "key";
 pub(crate) const MAP_VALUE_DEFAULT: &str = "value";
 
-impl TryFrom<&StructType> for ArrowSchema {
-    type Error = ArrowError;
+/// Convert a kernel type into an arrow type (automatically implemented for all types that
+/// implement [`TryFromKernel`])
+pub trait TryIntoArrow<ArrowType> {
+    fn into_arrow(self) -> Result<ArrowType, ArrowError>;
```

Comment:
```suggestion
    fn try_into_arrow(self) -> Result<ArrowType, ArrowError>;
```

I think we should stick with rust's naming convention that if an into/from is fallible it's a _try_into/from_